instrumentation/jaxrs-client/jaxrs-client-2.0/jaxrs-client-2.0-jersey-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrsclient/v2_0/JerseyClientUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jaxrsclient.v2_0;
+
+import static io.opentelemetry.javaagent.instrumentation.jaxrsclient.v2_0.JaxRsClientTracer.tracer;
+
+import io.opentelemetry.context.Context;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import org.glassfish.jersey.client.ClientRequest;
+
+public final class JerseyClientUtil {
+
+  public static Future<?> addErrorReporting(ClientRequest context, Future<?> future) {
+    // since jersey 2.30 jersey internally uses CompletableFuture
+    // we can't wrap it with WrappedFuture as it causes ClassCastException when casting
+    // to CompletableFuture
+    if (future instanceof CompletableFuture) {
+      future =
+          ((CompletableFuture<?>) future)
+              .whenComplete(
+                  (result, exception) -> {
+                    if (exception != null) {
+                      handleException(context, exception);
+                    }
+                  });
+    } else {
+      if (!(future instanceof WrappedFuture)) {
+        future = new WrappedFuture<>(future, context);
+      }
+    }
+
+    return future;
+  }
+
+  public static void handleException(ClientRequest context, Throwable exception) {
```

Comment:
We generally add instrumentation-specific handling to the `*Tracer` classes themselves instead of separating utils, would that work here?

Reply:
Tracer class is in a different module `jaxrs-client-2.0-common` adding these methods there would also require moving `WrappedFuture` there. Adding a tracer to `jaxrs-client-2.0-jersey-2.0` also doesn't seem like a good option because this module doesn't really have any specialised tracing code.